sofar/sofastream.py

```diff
@@ -0,0 +1,192 @@
+    --------
+    Get an attribute from a SOFA-file:
+
+        >>> from sofar import SofaStream
+        >>> filename = "path/to/file.sofa"
+        >>> with SofaStream(filename) as file:
+        >>>     data = file.GLOBAL_RoomType
+        >>>     print(data)
+        free field
+
+    Get a variable from a SOFA-file:
+
+        >>> with SofaStream(filename) as file:
+        >>>     data = file.Data_IR
+        >>>     print(data)
+        <class 'netCDF4._netCDF4.Variable'>
+        float64 Data.IR(M, R, N)
+        unlimited dimensions:
+        current shape = (11950, 2, 256)
+        filling on, default _FillValue of 9.969209968386869e+36 used
+
+    What is returned is a `netCDF-variable`. To access the values (in this
+    example the IRs) the variable needs to be sliced:
+
+        >>> with SofaStream(filename) as file:
+        >>>     data = file.Data_IR
+        >>>     # get all values
+        >>>     all_irs = data[:]
+        >>>     print(all_irs.shape)
+        (11950, 2, 256)
+        >>>     # get data from first channel
+        >>>     specific_irs = data[:,0,:]
+        >>>     print(specific_irs.shape)
+        (11950, 256)
+    """
+
+    def __init__(self, filename):
+        self._filename = filename
+
+    def __enter__(self):
+        self._file = Dataset(self._filename, mode="r")
+        return self
+
+    def __exit__(self, *args):
+        self._file.close()
+
+    def __getattr__(self, name):
+        # get netCDF4-attributes and -variable-keys from SOFA-file
+        dset_variables = np.array([key for key in self._file.variables.keys()])
+        dset_attributes = np.asarray(self._file.ncattrs())
+
+        # remove delimiter from passed sofar-attribute
+        name_netcdf = name.replace(
+            'GLOBAL_', '').replace('Data_', 'Data.')
+
+        # Handle variable-attributes (e.g. '_Units' and '_Type')
+        var_attr = None
+        if "_Units" in name_netcdf or "_Type" in name_netcdf:
+            name_split = name_netcdf.split('_')
+            name_netcdf = name_split[0]
+            var_attr = name_split[1]
+
+        # get value if passed attribute points to a netCDF4-variable
+        if name_netcdf in dset_variables:
+            # get variable from SOFA-file
+            self._data = self._file.variables[name_netcdf]
+            if var_attr is not None:
+                self._data = getattr(self._data, var_attr)
+
+        # get value if passed attribute points to a netCDF4-attribute
+        elif name_netcdf in dset_attributes:
+            # get attribute value from SOFA-file
+            self._data = self._file.getncattr(name_netcdf)
+
+        else:
+            raise ValueError("attribute is not in dataset")
+
+        return self._data
+
+    def inspect(self, file=None):
+        """
+        Get information about the data inside a SOFA-file
+
+        Prints all information about attributes, variables and their
+        shape, dimensions and values that are contained in a SOFA-file.
```

Comment:
Maybe use the same docstring as in Sofa.inspect()
```suggestion
        Prints the values of all attributes and variables with six or less
        entries and the shapes and type of all numeric and string variables.
        When printing the values of arrays, single dimensions are discarded for
        easy of display, i.e., an array of shape (1, 3, 2) will be displayed as
        an array of shape (3, 2).
```

Reply:
I don't understand why we simplify the shape form (1,2,3) to (2,3). I think its not related to this pr. Is it just to display or is it also simplified on the data itself

Reply:
I think that is bad decision that I made when I was young and wanted things to be as in Matlab